packages/contentful-migrations/core/0000000000003-create-t-page-migration.js

```diff
@@ -0,0 +1,245 @@
+    .localized(true)
+    .required(false)
+    .validations([])
+    .disabled(false)
+    .omitted(false)
+    .items({
+      type: 'Link',
+      validations: [
+        {
+          linkContentType: ['m-columns', 'm-editorial', 'm-list', 'm-text'],
+        },
+      ],
+      linkType: 'Entry',
+    });
+
+  tPage.changeFieldControl('internal_name', 'builtin', 'singleLine', {
+    helpText: 'e.g. "Home page"',
+  });
+
+  tPage.changeFieldControl('title', 'builtin', 'singleLine', {});
+
+  tPage.changeFieldControl('parent_page', 'builtin', 'entryLinkEditor', {});
+
+  tPage.changeFieldControl('slug', 'builtin', 'slugEditor', {
+    trackingFieldId: 'title',
+  });
+
+  tPage.changeFieldControl('seo_title', 'builtin', 'singleLine', {
+    helpText: 'This will override the page title in search engine results',
+  });
+
+  tPage.changeFieldControl('seo_description', 'builtin', 'singleLine', {
+    helpText: 'This will be displayed in search engine results',
+  });
+
+  tPage.changeFieldControl('share_image', 'builtin', 'assetLinkEditor', {
+    helpText: 'This will be displayed when sharing the page on social media',
+  });
+
+  tPage.changeFieldControl('no_index', 'builtin', 'boolean', {
+    helpText: 'Search engines will not include this page in search results',
+    trueLabel: 'Yes',
+    falseLabel: 'No',
+  });
+
+  tPage.changeFieldControl('no_follow', 'builtin', 'boolean', {
+    helpText: 'Search engines will not follow the links on your page',
+    trueLabel: 'Yes',
+    falseLabel: 'No',
+  });
+
+  tPage.changeFieldControl('theme', 'builtin', 'radio', {});
+
+  tPage.changeFieldControl('stage', 'builtin', 'entryLinkEditor', {});
+
+  tPage.changeFieldControl('modules', 'builtin', 'entryLinksEditor', {});
+
+  // Create editor layout (Compose only)
+  const editorLayout = tPage.createEditorLayout();
+
+  editorLayout.createFieldGroup('content', {
+    name: 'Content'
+  });
+
+  editorLayout.changeFieldGroupControl('content', 'builtin', 'topLevelTab', {
+    helpText: 'Main content'
+  });
+
+  editorLayout.createFieldGroup('settings').name('Settings');
+
+  editorLayout.changeFieldGroupControl('settings', 'builtin', 'topLevelTab', {
+    helpText: 'Page settings'
+  });
+
+  editorLayout.editFieldGroup('settings').createFieldGroup('seo').name('SEO');
+
+  editorLayout.changeFieldGroupControl('seo', 'builtin', 'fieldset', {
+    helpText: 'Search related fields',
+    collapsedByDefault: false
+  });
+
+  editorLayout.moveField('title').toTheTopOfFieldGroup('settings');
+
+  editorLayout.moveField('slug').afterField('title');
+
+  editorLayout.moveField('parent_page').afterField('slug');
+
+  editorLayout.moveField('seo_title').toTheTopOfFieldGroup('seo');
+
+  editorLayout.moveField('seo_description').afterField('seo_title');
+
+  editorLayout.moveField('share_image').afterField('seo_description');
+
+  editorLayout.moveField('no_index').afterField('share_image');
+
+  editorLayout.moveField('no_follow').afterField('no_index');
```

Comment:
do we also need some kind of `page` flag for compose?

Reply:
So that `t-page` is directly configured as Page Type in Compose? I haven't found anything in the migration docs. It didn't work via "fetch" either.  🤷🏻‍♂️

Or do you mean something else?

Reply:
Pages get the AggregateRoot annotation:
`.setAnnotations(['Contentful:AggregateRoot']);`
See https://www.contentful.com/developers/docs/compose/cli-tools/

There's another annotation to be used with components which directly belong to one page:
`.setAnnotations(['Contentful:AggregateComponent'])`
This "checks" the 'Mark as Component in Compose' checkbox.
https://www.contentful.com/developers/docs/compose/page-types/#when-to-activate-the-component-setting

Reply:
Done